src/lib/utils/meta_table_manager.cpp

```diff
@@ -96,18 +103,53 @@ MetaTableManager::MetaTableManager() {
   _methods["chunk_sort_orders"] = &MetaTableManager::generate_chunk_sort_orders_table;
   _methods["segments"] = &MetaTableManager::generate_segments_table;
   _methods["segments_accurate"] = &MetaTableManager::generate_accurate_segments_table;
+  _methods["plugins"] = &MetaTableManager::generate_plugins_table;
+
+  //const auto a = std::make_tuple(&MetaTableManager::_insert_into_plugins, &MetaTableManager::_delete_from_plugins);
+  //auto a =  {&MetaTableManager::_insert_into_plugins, &MetaTableManager::_delete_from_plugins};
+  tuple<std::function<std::shared_ptr<Table>(const std::string&)>,
+        std::function<std::shared_ptr<Table>(const std::string&)>>(&MetaTableManager::_insert_into_plugins,
+                                                                   &MetaTableManager::_delete_from_plugins) a;
+
+  _mutating_methods["plugins"] = a;
```

Comment:
Change `_methods["..."]` to hold a class that has methods for get, insert, and delete (which may default to something throwing if nothing is implemented)? That way, we can move those classes into separate files and the MetaTableManager itself doesn't have to know anything about plugins.

Reply:
The SQLTranslator could check if a method (e.g., INSERT) is set and throw an input exception if it is not.

Also, we probably should discuss how to handle update. DELETE+INSERT doesn't feel right as we don't have proper MVCC on the meta tables and shouldn't leave the DB in an inconsistent state

Reply:
> That way, we can move those classes into separate files and the MetaTableManager itself doesn't have to know anything about plugins.

Okay, do you have any suggestions where those classes should be located at best? `utils/meta_table_manipulation`?

> DELETE+INSERT doesn't feel right as we don't have proper MVCC on the meta tables and shouldn't leave the DB in an inconsistent state

MVCC with meta tables is something that also came to my mind. Is this a general thing that should be thought of? I mean, since tey are generated on the fly and will be wrapped in a static table node, does having MVCC here really makes sense?

Reply:
> Okay, do you have any suggestions where those classes should be located at best? utils/meta_table_manipulation?

Just `utils/meta_tables`? Have a class for each meta table and that class either implements modifications or not.

> MVCC with meta tables is something that also came to my mind. Is this a general thing that should be thought of?

Oh god, no. Quite the opposite - I was trying to say that our UPDATE=DELETE+INSERT concept might be a disadvantage here.